src/sentry/static/sentry/app/views/organizationDiscover/index.jsx

```diff
@@ -58,7 +59,7 @@ const OrganizationDiscoverContainer = createReactClass({
 
   componentWillReceiveProps: function(nextProps) {
     if (!nextProps.params.savedQueryId) {
-      this.setState({savedQuery: null});
+      this.setState({savedQuery: null, isEditingSavedQuery: false});
```

Comment:
Is it possible to transition from a saved query with `editing=true` to not having a `savedQueryId` with `editing=true`? It seems like `isEditingSavedQuery` doesn't need to be in state since it's always equal to `this.props.location.query.editing`?

Reply:
Yeah, we can remove this state variable